### DISCUSSION TITLE
Clarification about ports in docker

### DISCUSSION THREAD

**Discussion by markwilkinson** [bulletmark/corsproxy#7] [Announcements]
Hi!

Your instructions for Docker are a bit... opaque...

$ docker run --restart always -d -p 8000:8000 bulletmark/corsproxy 8000=http://192.168.1.98

It appears that your software, inside of the container, is listening on port 8000 - is that correct?  So does the 8000 in the argument refer to the external 8000 port, or the internal 8000 port?

If i wanted to receive requests on port 5000 and forward them to 192.168.1.1:9000, what would the equivalent docker command look like?  I'm having problems understanding when everything is 8000 :-)

Cheers!

**Comment by bulletmark** [comment]
I converted this to a discussion since you are asking a question, not raising a bug/issue.

Your first question is asking about the format of docker command line argument, specifically about how ports are published. Refer to the [docker documentation](https://docs.docker.com/network/#published-ports) which is pretty clear.

Your second question is whether the `8000=` argument refers to the internal or external port. The `corsproxy` software is running within the container and can't know about anything external so it's given `8000=` argument is the internal port that it can see, i.e. within the environment where that software is running.

Your third question is about how to receive on port `5000` and forward to port `9000`. The `corsproxy` documentation and examples show the format of the argument is `[host:]port=http://server[:targetport]`. The following command should do what you want.

    $ docker run --restart always -d -p 5000:8000 bulletmark/corsproxy 8000=http://192.168.1.1:9000

I.e. the `8000` in both places above can be anything really. It's just a hidden internal port we use to plug through into the container. I'd probably use `5000` for that too but you may not like that ;)
